shift_detector/preprocessors/NGram.py

```diff
@@ -1,29 +1,43 @@
-class NGram:
+from shift_detector.preprocessors.Preprocessor import Preprocessor
+from enum import Enum
+import pandas as pd
 
-    def __init__(self, n=5):
-        self.n = n
-        if self.n < 1:
-            raise Exception('n has to be greater than 0')
 
-    def __eq__(self, other):
-        """Overrides the default implementation"""
-        return self.n == other.n
+class NGramType(Enum):
+    word = "word"
+    character = "character"
 
-    def __hash__(self):
-        return hash((self.__class__, self.n))
 
-    def process(self, train, test):
+class NGram(Preprocessor):
 
-        def generate_ngram(text, n):
-            ngram = {}
-            for i in range(len(text) - n + 1):
-                ngram[text[i:i+n]] = 1 if text[i:i+n] not in ngram else ngram[text[i:i+n]] + 1
-            return ngram
+    def __init__(self, n: int, ngram_type : NGramType):
+        self.n = n
+        self.ngram_type = ngram_type
+        if self.n < 1:
+            raise ValueError('n has to be greater than 0')
 
-        train = train.dropna().str.lower()
-        train_processed = train.apply(lambda row: generate_ngram(row, self.n))
+    def __eq__(self, other):
+        return self.n == other.n and self.ngram_type == other.ngram_type
 
-        test = test.dropna().str.lower()
-        test_processed = test.apply(lambda row: generate_ngram(row, self.n))
+    def __hash__(self):
+        return hash((self.__class__, self.n, self.ngram_type))
+
+    def generate_ngram(self, text: tuple) -> dict:
+        ngram = {}
+        for i in range(len(text) - self.n + 1):
+            ngram[tuple(text[i:i + self.n])] = 1 if tuple(text[i:i + self.n]) not in ngram \
+                else ngram[tuple(text[i:i + self.n])] + 1
+        return ngram
+
+    def process(self, train: pd.Series, test: pd.Series) -> tuple:
+        if self.ngram_type == NGramType.word:
```

Comment:
I think it would be more understandable to use:
```python
train = train.dropna().str.lower()
test = test.dropna().str.lower()

if self.ngram_type == NGramType.word:
    train = train.str.split()
    test = test.str.split()           
```